.github/workflows/nightly.yaml

```
@@ -0,0 +1,133 @@
+          echo 'REDHAT_STANDARD<<EOF' >> $GITHUB_ENV
+          echo 'type=raw,value=nightly,suffix=-redhat' >> $GITHUB_ENV
+          echo "type=raw,value={{date 'YYYY-MM-DD'}},suffix=-redhat" >> $GITHUB_ENV
+          echo 'EOF' >> $GITHUB_ENV
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2.0.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Login to RH Scan Registry 
+        uses: docker/login-action@v2
+        with:
+          registry: scan.connect.redhat.com
+          username: ${{ secrets.RH_USERNAME }}
+          password: ${{ secrets.RH_TOKEN }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4.0.1
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: ${{ env.TAGS_STANDARD }}
+
+      - name: Docker meta (RedHat)
+        id: meta_redhat
+        uses: docker/metadata-action@v4.0.1
+        with:
+          images: ${{ env.RH_SCAN_REGISTRY }}/${{ env.RH_IMAGE_NAME }}
+          flavor: |
+            latest=false
+          tags: ${{ env.REDHAT_STANDARD }}
+
+      - name: Build binary
+        id: docker_build_binary
+        uses: docker/build-push-action@v3
+        with:
+          push: false
+          file: Dockerfile
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache
+          target: builder
+          platforms: linux/amd64, linux/arm64
+          build-args: |
+            TAG=${{ steps.meta.outputs.version }}
+            COMMIT=${{ github.sha }}
+            REPO_INFO=https://github.com/${{ github.repository }}.git
+          
+      - name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          file: Dockerfile
+          tags: ${{ steps.meta.outputs.tags }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          target: distroless
+          platforms: linux/amd64, linux/arm64
+          build-args: |
+            TAG=${{ steps.meta.outputs.version }}
+            COMMIT=${{ github.sha }}
+            REPO_INFO=https://github.com/${{ github.repository }}.git
+
+      - name: Build and push Red Hat
+        id: docker_build_redhat
+        env:
+          TAG: ${{ steps.meta_redhat.outputs.version }}
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          file: Dockerfile
+          tags: ${{ steps.meta_redhat.outputs.tags }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          target: redhat
+          # RedHat scan registry does not support multiarch images
+          # platforms: linux/amd64, linux/arm64
+          platforms: linux/amd64
```

Right this is what we discussed yesterday. I don't think we need to leave the comment with the multiple platforms in:

```suggestion
          # RedHat scan registry does not support multiarch images
          platforms: linux/amd64
```

However, what I DO think we should add here is a link to the relevant Redhat documentation which describes this limitation, that way future readers have some breadcrumbs to work with if they're trying to make changes to the platform support for this workflow.

---

There is no doc entry for that. I've learned by experimenting and googling. scan registry refused to accept the multiarch image with internal server error 500, with no additional comments. By googlong I've learned that it's common problem, not all image registries support multiarch yet.

---

Well, I can't say I'm particularly surprised that they don't document this, it just fails. Ok then :shrug: